test/unit/test_analysis_global_var.c

```diff
@@ -0,0 +1,115 @@
+// SPDX-FileCopyrightText: 2021 Basstorm <basstorm@nyist.edu.cn>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_analysis.h>
+#include "minunit.h"
+
+bool test_rz_analysis_global_var() {
+	RzAnalysis *analysis = rz_analysis_new();
+
+	RzAnalysisVarGlobal *glob = rz_analysis_var_global_new("foo", 0x1337);
+	mu_assert_notnull(glob, "create a global variable");
+	mu_assert_streq(glob->name, "foo", "global var name");
+	mu_assert_eq(glob->addr, 0x1337, "global var address");
+	RzTypeParser *parser = rz_type_parser_new();
+	mu_assert_notnull(parser, "create type parser");
+	char *errmsg = NULL;
+	RzType *typ = rz_type_parse_string_single(parser, "int", &errmsg);
+	mu_assert_notnull(typ, "parsed type");
+	rz_analysis_var_global_set_type(glob, typ);
+	mu_assert_streq(glob->type->identifier.name, "int", "global var type");
+
+	bool added = rz_analysis_var_global_add(analysis, glob);
+	mu_assert_true(added, "add global var");
+
+	glob = NULL;
+	glob = rz_analysis_var_global_get_byaddr(analysis, 0x1337);
+	mu_assert_notnull(glob, "get global var by addr");
+	mu_assert_streq(glob->name, "foo", "global var name");
+	mu_assert_eq(glob->addr, 0x1337, "global var address");
+	mu_assert_streq(glob->type->identifier.name, "int", "global var type");
+
+	glob = NULL;
+	glob = rz_analysis_var_global_get_byname(analysis, "foo");
+	mu_assert_notnull(glob, "get global var by addr");
+	mu_assert_streq(glob->name, "foo", "global var name");
+	mu_assert_eq(glob->addr, 0x1337, "global var address");
+	mu_assert_streq(glob->type->identifier.name, "int", "global var type");
+
+	glob = NULL;
+	glob = rz_analysis_var_global_get_byaddr(analysis, 0x1339); // test RBTree
+	mu_assert_notnull(glob, "get global var by addr");
+	mu_assert_streq(glob->name, "foo", "global var name");
+	mu_assert_eq(glob->addr, 0x1337, "global var address");
+	mu_assert_streq(glob->type->identifier.name, "int", "global var type");
+
+	bool rename = rz_analysis_var_global_rename(analysis, "foo", "bar");
+	mu_assert_true(rename, "rename global var");
+	glob = NULL;
+	glob = rz_analysis_var_global_get_byname(analysis, "bar");
+	mu_assert_notnull(glob, "get global var by addr");
+	mu_assert_streq(glob->name, "bar", "global var name");
+
+	bool deleted = rz_analysis_var_global_delete_byaddr(analysis, 0x1337);
+	mu_assert_true(deleted, "delete global var");
+	glob = NULL;
+	glob = rz_analysis_var_global_get_byaddr(analysis, 0x1337);
+	mu_assert_null(glob, "get deleted global var");
+
+	// re add
```

Comment:
just a note, no need to change it now or at all, but I think it is usually better to have smaller unit tests that just do few things instead of a single one that does many things together.